x/cronos/types/messages.go

```diff
@@ -171,3 +176,32 @@ func (msg *MsgUpdateTokenMapping) GetSignBytes() []byte {
 	bz := ModuleCdc.MustMarshalJSON(msg)
 	return sdk.MustSortJSON(bz)
 }
+
+func NewMsgUpdateParams(authority string, params Params) *MsgUpdateParams {
+	return &MsgUpdateParams{
+		Authority: authority,
+		Params:    params,
+	}
+}
+
+// GetSigners returns the expected signers for a MsgUpdateParams message.
+func (msg *MsgUpdateParams) GetSigners() []sdk.AccAddress {
+	addr, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{addr}
+}
```

Comment:
```suggestion
func getSigners(adr string) []sdk.AccAddress {
	address, err := sdk.AccAddressFromBech32(adr)
	if err != nil {
		panic(err)
	}
	return []sdk.AccAddress{address}
}

// GetSigners returns the expected signers for a MsgUpdateParams message.
func (msg *MsgUpdateParams) GetSigners() []sdk.AccAddress {
	return getSigners(msg.Authority)
}
```

Reply:
could apply for rest `GetSigners()`

Reply:
is this better design pattern?

Reply:
just simple reuse, since 4 `GetSigners` already